generators/server/index.js

```diff
@@ -205,9 +209,12 @@ module.exports = JhipsterServerGenerator.extend({
     prompting: {
 
         askForModuleName: function () {
-            if (this.baseName) return;
+            //need this for having a concrete name in gateway app to build urls like "/jhipster-uaa/api/account"
+            // @TODO read this application name from gateway during its generation
+            if (this.baseName || this.applicationType == 'uaa') return;
```

Comment:
`this.applicationType == 'uaa'` check seems redundant as its done in the following method already, can we keep it either here or in method only